### ISSUE TITLE
Array.map is slow in bun.js

### ISSUE THREAD

**Issue by Bhardwaj-Raghav** [oven-sh/bun#1096]
### Version

0.1.8

### Platform

Linux raghav 5.15.0-46-generic #49-Ubuntu SMP Thu Aug 4 18:03:25 UTC 2022 x86_64 x86_64 x86_64 GNU/Linux

### What steps will reproduce the bug?

Doing simple array map in js file. 
![image](https://user-images.githubusercontent.com/47305069/185346007-6e299d90-3ac2-4fc0-940e-6cfc27d649de.png)

### How often does it reproduce? Is there a required condition?

Every time, No
![image](https://user-images.githubusercontent.com/47305069/185346527-0ac27cca-b71b-4b25-b8eb-057814958596.png)

### What is the expected behavior?

For the Fastest JavaScript runtime, expectation is that is will be faster than Node.js

### What do you see instead?

Array.map is slower in bun.js then node.js

**Comment by paperclover** [comment]
i can confirm that this specific type of thing is slower on bun. here's a benchmark using `mitata` with a "no map" test as a baseline.
![image](https://user-images.githubusercontent.com/24465214/185556049-e00a08c4-0dd8-4c00-b562-a0a49bf82a39.png)

```ts
import { bench, run } from "mitata";

bench("array.map", () => {
  const array = ["A", "B", "C", "D", "E", "F", "G", "H", "I", "J", "K", "L", "M", "N", "O", "P", "Q", "R", "S", "T", "U", "V", "W", "X", "Y", "Z", "A", "B", "C", "D", "E", "F", "G", "H", "I", "J", "K", "L", "M", "N", "O", "P", "Q", "R", "S", "T", "U", "V", "W", "X", "Y", "Z", "A", "B", "C", "D", "E", "F", "G", "H", "I", "J", "K", "L", "M", "N", "O", "P", "Q", "R", "S", "T", "U", "V", "W", "X", "Y", "Z", "A", "B", "C", "D", "E", "F", "G", "H", "I", "J", "K", "L", "M", "N", "O", "P", "Q", "R", "S", "T", "U", "V", "W", "X", "Y", "Z", "A", "B", "C", "D", "E", "F", "G", "H", "I", "J", "K", "L", "M", "N", "O", "P", "Q", "R", "S", "T", "U", "V", "W", "X", "Y", "Z", "A", "B", "C", "D", "E", "F", "G", "H", "I", "J", "K", "L", "M", "N", "O", "P", "Q", "R", "S", "T", "U", "V", "W", "X", "Y", "Z"];
  return array.map((x) => x);
});

run({ percentiles: false });
```

**Comment by Jarred-Sumner** [comment]
I've filed a bug with WebKit/JavaScriptCore about this. Most of the gap here is when using `.map` on an array literal. It smells like a JIT optimization that exists in V8 which JSC hasn't implemented yet. JSC's Array.prototype.map implementation is in JavaScript.

https://bugs.webkit.org/show_bug.cgi?id=245797

**Comment by Jarred-Sumner** [comment]
Fixed in Bun v0.2.0

<img width="384" alt="image" src="https://user-images.githubusercontent.com/709451/194950686-3eb82bd2-199b-4e6a-ad3d-1ef21f6d5c8c.png">
